Update py-openstackclient to 6.2.0 + dependencies

PixPMusic opened the pull request:
#### Description
openstackclient was horribly out of date, and the `--tag` flag was bugged. Newer versions fix this issue, so I've updated the portfiles related.
I was unable to get port checksum to generate checksums even after a build, so those will need to be generated and added to these portfiles.
*most* are nomaintainer, but a couple have maintainers, but I'm only enabling newer Python builds on those as far as I can tell.

py-cinderclient: 7.1.0 => 9.3.0
py-cliff: 3.3.0 => 4.3.0
py-cmd2: 2.3.3 => 2.4.3
py-debtcollector: 2.1.0 => 2.5.0
py-dogpile-cache: 1.0.1 => 1.2.2
py-keystoneauth1: 4.2.0 => 5.2.1
py-keystoneclient: 4.1.0 => 5.1.0
py-novaclient: 17.2.0 => 18.3.0
py-openstackclient: 5.3.1 => 6.2.0
py-openstacksdk: 0.48.0 => 1.4.0
py-osc-lib: 2.2.0 => 2.8.0
py-oslo-config: 8.3.1 => 9.1.1
py-oslo-i18n: 5.0.0 => 6.0.0
py-oslo-serialization: 4.0.0 => 5.1.1
py-oslo-utils: 4.3.0 => 6.2.0

Updated minimum Python versions to match PyPi listed for all above, and enabled all to be installed on up to 3.11
py-requestsexceptions add py35 py36 py39 py310 py311 py-os-service-types add py39 py310 py311
py-netifaces, py-prettytable add py311

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 13.5 22G74 arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

reneeotten commented on python/py-cinderclient/Portfile:
```diff
@@ -22,11 +22,7 @@ long_description    Client for the OpenStack Volume API. There’s a \
 
 homepage            https://docs.openstack.org/python-cinderclient/latest/
 
-checksums           rmd160  9706fd47624c8490cfbbcc8620a4fe986ecb79b0 \
```
you cannot just remove the `checksums`, instead they should be updated

reneeotten commented on python/py-cliff/Portfile:
```diff
@@ -16,11 +16,8 @@ long_description    cliff is a framework for building command line \
                     provide subcommands, output formatters, and other \
                     extensions.
 homepage            https://docs.openstack.org/cliff/latest/
-checksums           rmd160  131952d0886ea27d1b4225c3065c2ee5467cc63d \
```
cannot be removed

reneeotten commented on python/py-cmd2/Portfile:
```diff
@@ -16,11 +16,7 @@ description         Tool for building interactive command line applications in P
 long_description    {*}${description}
 homepage            https://github.com/python-cmd2/cmd2
 
-checksums           rmd160  8f15ccfc87a3b094b68682da668e1562ce819cf5 \
```
cannot be removed

reneeotten reviewed (changes requested):
@PixPMusic thanks for the work on updating all these ports! 

However, this can never work as for several ports where you update the version you either completely remove the `checksums` or don't update them. 

Additionally, you remove/add Python subports here and there. It's okay to add newer ones, but adding `35` or `36` is not what we want since that is already an EOL Python version. Additionally, when removing subport you'll need to verify that these do not have dependents anymore - did you do that?

PixPMusic commented:
@reneeotten I did mention in my PR:
> I was unable to get port checksum to generate checksums even after a build, so those will need to be generated and added to these portfiles.
I'll gladly add these checksums if I could get them to generate?

As for separate commits, can do.

As for analyzing and testing every dependency, that's obviously more challenging. My focus was checking the lowest supported version of each library on PyPi and I updated the Portfiles accordingly. If this is not something that MacPorts would like, due to the EoL nature of some of these versions of Python, I understand.

I think a better focus here would be if my commits didn't change some of these library versions, but only enable newer versions of Python. My end goal was just to get OpenStack's CLI working.

reneeotten commented:
> I'll gladly add these checksums if I could get them to generate?

"sudo port -dv checksums pyXY-portname"

Additionally, you also should double check with the checksums published on PyPI. 

> As for separate commits, can do.

Thanks!

> As for analyzing and testing every dependency, that's obviously more challenging. My focus was checking the lowest supported version of each library on PyPi and I updated the Portfiles accordingly. If this is not something that MacPorts would like, due to the EoL nature of some of these versions of Python, I understand.

Yes, just don't add subports EOL/older Python versions. And if you are removing subports, just check if other ports depend in them:

"port echo depends:pyXY-portname"

reneeotten commented:
ping @PixPMusic ?

pmetzger commented:
@PixPMusic It would be nice to get this finished. Do you think you could make the need final fixes?

PixPMusic commented:
Since I'm actually doing this for work, I'm going to go ahead and open a PR against my work account. Closing this out. Sorry for the bother

pmetzger commented:
I'll comment on the new PR.
